### ISSUE TITLE
Bug Report: Query with exists and CHAR column type throw an error

### ISSUE THREAD

**Issue by L3o-pold** [vitessio/vitess#10095]
### Overview of the Issue

An error occurs when doing an EXISTS sub query on `CHAR` MySQL type. 

```
Type is not supported: "SKU-1001" CHAR
```

### Reproduction Steps

1. Deploy the following vschema:

```json
{
    "sharded": true,
    "vindexes": {
        "binary": {
            "type": "binary"
        },
        "unicode_loose_md5": {
            "type": "unicode_loose_md5"
        }
    },
    "tables": {
        "customer": {
            "column_vindexes": [
                {
                    "column": "customer_id",
                    "name": "binary"
                }
            ]
        },
        "corder": {
            "column_vindexes": [
                {
                    "column": "customer_id",
                    "name": "binary"
                }
            ]
        },
        "product": {
            "column_vindexes": [
                {
                    "column": "sku",
                    "name": "unicode_loose_md5"
                }
            ]
        }
    }
}
```

2. Deploy the following `schema`

```sql
create table product
(
    sku         char(8) not null,
    description varbinary(128),
    price       bigint,
    primary key (sku)
) ENGINE=InnoDB;
create table customer
(
    customer_id binary(16) not null,
    email       varbinary(128),
    primary key (customer_id)
) ENGINE=InnoDB;
create table corder
(
    order_id    binary(16) not null,
    customer_id binary(16) not null,
    sku         char(8),
    price       bigint,
    primary key (order_id)
) ENGINE=InnoDB;
```

3. Add some datas

```sql
set @a = unhex(replace(uuid(),'-',''));
insert into customer(customer_id, email) VALUES (@a, "alice@domain.com");
insert into product(sku, description, price) values('SKU-1001', 'Monitor', 100);

set @b = unhex(replace(uuid(),'-',''));
insert into corder(order_id, customer_id, sku, price) values(@b, @a, 'SKU-1001', 100);
```

4. Run the query

```sql
select corder.`sku` 
from `corder` 
where `sku` = 'SKU-1001'
and exists (
	select `product`.`sku` 
	from `product` 
	where `corder`.`sku` = `product`.`sku` 
	and `product`.`description` = "Monitor");
```

### Binary Version

```sh
Version: 13.0.0 (Git revision bc4a9606e1 branch 'heads/v0.13.0') built on Tue Feb 22 14:37:16 UTC 2022 by vitess@buildkitsandbox using go1.17.6 linux/amd64
```

### Operating System and Environment details

```sh
Docker
```

**Comment by L3o-pold** [comment]
Same thing occurs when replacing `sku` type from `char` to `binary`.

**Comment by mattlord** [comment]
Hi @L3o-pold !

It seems like this is fixed in `main`/`14.0.0-SNAPSHOT`:
```
git checkout main && make docker_local && ./docker/local/run.sh

./201_customer_tablets.sh && ./202_move_tables.sh && ./203_switch_reads.sh && sleep 5 && ./204_switch_writes.sh && ./205_clean_commerce.sh

./301_customer_sharded.sh && ./302_new_shards.sh && ./303_reshard.sh && sleep 5 && ./304_switch_reads.sh && ./305_switch_writes.sh && ./306_down_shard_0.sh && ./307_delete_shard_0.sh

mysql customer -e "alter table corder modify sku char(8); alter table customer modify customer_id binary(16) not null; alter table corder modify order_id binary(16) not null, modify customer_id binary(16) not null; create table product (sku char(8) not null, description varbinary(128), price  bigint, primary key (sku)) ENGINE=InnoDB;"

cat <<EOF>/tmp/vschema.json
{
    "sharded": true,
    "vindexes": {
        "binary": {
            "type": "binary"
         },
         "unicode_loose_md5": {
             "type": "unicode_loose_md5"
         }
    },
    "tables": {
        "customer": {
            "column_vindexes": [
                {
                    "column": "customer_id",
                    "name": "binary"
                }
            ]
        },
        "corder": {
            "column_vindexes": [
                {
                    "column": "customer_id",
                    "name": "binary"
                }
            ]
        },
        "product": {
            "column_vindexes": [
                {
                    "column": "sku",
                    "name": "unicode_loose_md5"
                }
            ]
        }
    }
}
EOF

vtctlclient ApplyVschema -vschema_file=/tmp/vschema.json customer

mysql customer -e "set @a = unhex(replace(uuid(),'-','')); insert into customer(customer_id, email) VALUES (@a, 'alice@domain.com'); insert into product(sku, description, price) values('SKU-1001', 'Monitor', 100); set @b = unhex(replace(uuid(),'-','')); insert into corder(order_id, customer_id, sku, price) values(@b, @a, 'SKU-1001', 100);"
```

And the final results:
```
mysql -e "select corder.sku from corder where sku = 'SKU-1001' and exists (select product.sku from product where corder.sku = product.sku and product.description = 'Monitor')"
+----------+
| sku      |
+----------+
| SKU-1001 |
+----------+

vtgate --version
ERROR: logging before flag.Parse: E0414 17:43:52.045149    8366 syslogger.go:149] can't connect to syslog
Version: 14.0.0-SNAPSHOT (Git revision 1e35237574cccb2dc9564a9fb9e85e7f7f35b2a8 branch 'main') built on Thu Apr 14 16:55:28 UTC 2022 by vitess@00b7cb12dc7f using go1.18 linux/amd64
```

Am I missing something? Thanks!

**Comment by L3o-pold** [comment]
Thanks for your quick feedback @mattlord. I confirm there is no issue in `main/14.0.0-SNAPSHOT`. Sorry for the noise.
